### ISSUE TITLE
How many steps for training?

### ISSUE THREAD

**Issue by caixd-220529** [siyan-zhao/OPSD#3]
Hi, thank for the great work, Can you share how many gradient steps are used for taining for the OPSD and GRPO?

**Comment by siyan-zhao** [comment]
Hi, thanks for your interest. For the updated results (as shown in blog & to be updated on Arxiv in few day), we only trained 100 gradient steps for OPSD and for GRPO we trained 500 steps. We evaluated every 20 steps up to 100 steps, as we find OPSD converges very fast within 100 steps and later steps can be unstable.  Hope this helps!

**Comment by caixd-220529** [comment]
@siyan-zhao Thanks for the quick response. It really helps. I still wondering here the steps means (a) the gradient update steps as mentioned in table 3 of the paper, OR (b) the actual training steps. Is this step configuration works well for model of other size (e.g., 4b, 8b)?

Also, I am also curious the training steps of the SFT baseline. Is it set to the same as `run_sft.sh`?

**Comment by siyan-zhao** [comment]
Hi! Please see our updated [paper’s](https://arxiv.org/pdf/2601.18734v3) configurations in Table2. Yes it is gradient update steps. And we did the same for 1B 4b 8b. 
For OPSD, we evaluate checkpoints every 20 steps up to 100 steps and report the best score. 
For GRPO, we report the peak performance
within 500 training steps, though we find GRPO performance to decrease for some tasks due to entropy collapse in later steps. For SFT,
we train on the same number of samples as OPSD. SFT performance degrades due to fine-tuning on concise reasoning solutions and
reduces generation length at test time.
